app/src/main/java/org/mozilla/fenix/components/TopSiteStorage.kt

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import mozilla.components.feature.top.sites.TopSite
+import mozilla.components.feature.top.sites.TopSiteStorage
+import org.mozilla.fenix.test.Mockable
+
+@Mockable
+class TopSiteStorage(private val context: Context) {
```

Comment:
Since this class seems to just duplicate the AC `TopSiteStorage`, why is it needed?

Reply:
Yea, this one is totally weird to me too. Right now, this is a wrapper around the AC `TopSiteStorage`, which will hold onto the reference to AC's `TopSiteStorage` and a local cache `cachedTopSites`. I don't know a better place for this so I am open to a better solution. For the time being, I will note it is following the existing pattern with `TabCollectionStorage`.